### ISSUE TITLE
Climatic > Prepare > Transform dialogue

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#2973]
The existing dialogue at that menu position is (I think) just a copy of a dialogue that is disabled, at Prepare > Column: Calculate > Transform.  It is (for now) OK in Prepare, but the dialogue needed in Climatic is completely different.

So start with a new dialogue. 
a) If you would like to base it on an existing dialogue then I suggest you could start with the Climatic > Prepare > Specific Events > Water Balance dialogue.  (but look at point i below, before starting the dialogue.)
b) This dialogue will be very simple.  It will take a climatic column and transform it slightly, producing another column of the same length, i.e. usually still daily.  Then it writes this new column back to the same data frame as the original (usually rainfall) data column.
c) If you want to understand what it is doing, then it is the same as in the "old Instat".  There the dialogues for the start of the rains, spells, and water balance each had a "simple" option, and this did the same thing.  I can demonstrate and explain if needed.
d) There is the usual **data selector**.  We assume also that (usually) the Climatic > Define Climatic Data has been used earlier.  So some of the following fields will be completed automatically.
e) On the right the first field has the label **"Station"**.  This isn't yet there in the other climatic dialogues, but it needs to be added.
f) The next fields - similar layout to the Water Balance dialogue - are Date, Year, Day of Year, Data in that order.
g) The only real difference in f) compared to the water balance dialogue is "**Data**" rather than "**Rain Column**".  If rain has been defined, then fill that field with the rain column.  But the dialogue will also work for temperatures for some of the transformations.  
h) Include the **Threshold** field, as in the Water Balance dialogue.
i) Here is where it gets different.  At the top of the dialogue there are 4 of our new "buttons", saying **Sum, Count, Spell, Water Balance**. 
j) Depending on which is chosen there is a block that opens, with the label being the same as the top.
k) If **Sum** then there is another set of buttons (or maybe this time radio buttons - check with Danny) , with the options Sum, Mean, Max, Min.  Then it says "**Over" then **up-down control** then " rows"**.  The up-down control is between 1 and 366, default is 3.  The default is Sum.
l) If Count is says "**Count over" then up-down control " rows"**.  There is then a checkbox, default checked which says "Values under threshold."
m) If Spell it says Values between <input field for number> and <input field for number> 
n) If Water it says Capacity and then the up-down control from the water balance.  And then Evaporation and a field to input just a number (for now!)
o) Below this box is the usual field to save the results.  The default name depends on what option was chosen.

The calculations should each be relatively simple examples of the new "calculation system".  This will need discussion with Danny and/or Lily.

**Comment by stevekogo** [comment]
@shadrackkibet You can go ahead and create this dialog(empty dialog) then we can work on it together.

**Comment by shadrackkibet** [comment]
Great

**Comment by stevekogo** [comment]
I am working on this

**Comment by rdstern** [comment]
This seems to have started and then has slowed.  Here are comments on what is there now - which is looking good:
here are 3 minor suggestions on what is there now.
a) Change "Sum" to "Moving", both in the choice at the top and also in the block below.
b) Change the label "Rain Column" to "Element".  Allow any element here, but if it finds Rain, then that is put there (as now) as the default.
c) Get rid of the wretched up-down control for the threshold and just have a number there, with 0.85 as the default.

**Comment by rdstern** [comment]
If you are going to calculate a spell, then it has to be a spell of values in a particular range.  For rainfall we have perhaps the default of between 0 and 0.85 (inclusive).  For temperatures it could be 30 to 50 degrees, etc.

**Comment by stevekogo** [comment]
Perfect! I will implement this right away.

**Comment by shadrackkibet** [comment]
@dannyparsons  this can be closed. @stevekogo worked on this and it was merged.

**Comment by dannyparsons** [comment]
I think there is still work to do on this so I'll keep it open in case the message here are useful.

**Comment by volloholic** [comment]
@dannyparsons I propose we close this and open a new issue with any remaining aspects so they can be assigned an appropriate milestone.

**Comment by dannyparsons** [comment]
Sounds fine, who will pick out the remaining issues?
